bundles/binding/org.openhab.binding.temperlan/src/main/java/org/openhab/binding/temperlan/internal/TemperlanBinding.java

```diff
@@ -0,0 +1,228 @@
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ * 
+ * @author snoerenberg
+ * @since 1.0
+ */
+public class TemperlanBinding extends AbstractActiveBinding<TemperlanBindingProvider> implements ManagedService {
+
+	public static final String CONFIG_KEY_HOST = "host";
+	public static final long DEFAULT_REFRESH_INTERVAL = 60000;
+	public static final String DEFAULT_DEVICE_UID = "default";
+
+	private static final String BINDING_NAME = "TemperlanBinding";
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(TemperlanBinding.class);
+
+	private long refreshInterval = DEFAULT_REFRESH_INTERVAL;
+
+	// Map of proxies. key=deviceUid, value=proxy
+	// Used to keep track of proxies
+	private final Map<String, TemperlanProxy> proxies = new HashMap<String, TemperlanProxy>();
+
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	@Override
+	protected String getName() {
+		return "Temperlan Refresh Service";
+	}
+
+	@Override
+	protected void execute() {
+		try {
+			// Iterate through all proxies
+			for (Map.Entry<String, TemperlanProxy> entry : proxies
+					.entrySet()) {
+				String deviceUid = entry.getKey();
+				TemperlanProxy receiverProxy = entry.getValue();
+				sendUpdates(receiverProxy, deviceUid);
+			}
+		} catch (Throwable t) {
+			logger.error("Error polling devices for " + getName(), t);
+		}
+	}
+
+	private void sendUpdates(TemperlanProxy receiverProxy, String deviceUid) {
+		// Get all item configurations belonging to this proxy
+		Collection<TemperlanBindingConfig> configs = getDeviceConfigs(deviceUid);
+		try {
+			
+				// Poll the state from the device
+				TemperlanState state = receiverProxy.getState();
+	
+				// Create state updates
+				State temperature1Update = new DecimalType(state.getTemperature1());
+				State temperature2Update = new DecimalType(state.getTemperature2());
+				State temperature3Update = new DecimalType(state.getTemperature3());
+				State temperature4Update = new DecimalType(state.getTemperature4());
+				State temperature5Update = new DecimalType(state.getTemperature5());
+				State temperature6Update = new DecimalType(state.getTemperature6());
+				State temperature7Update = new DecimalType(state.getTemperature7());
+				State temperature8Update = new DecimalType(state.getTemperature8());
+				State temperature9Update = new DecimalType(state.getTemperature9());
+				State temperature10Update = new DecimalType(state.getTemperature10());
+				State temperature11Update = new DecimalType(state.getTemperature11());
+				State temperature12Update = new DecimalType(state.getTemperature12());
+
+				// Send updates
+				if(state.getTemperature1() > -200) sendUpdate(configs, BindingType.temperature1, temperature1Update);

```

Comment:
change to a more oo-ish way :-) add a method like isValidTemperature1() to TemperlanState